From c67dd9d3fe9186bef76ba682a7e277ccb7d95227 Mon Sep 17 00:00:00 2001
From: Philip Withnall <pwithnall@endlessos.org>
Date: Tue, 18 Jul 2023 11:31:01 +0100
Subject: [PATCH] gmain: Add a missing return on error path in
 g_main_context_release()

This should have been in commit
3926af723a7469a2ea492307f421820361d617b3.

Signed-off-by: Philip Withnall <pwithnall@endlessos.org>

Fixes: #3054
---
 glib/gmain.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/glib/gmain.c b/glib/gmain.c
index fac14224a..ccadf76ce 100644
--- a/glib/gmain.c
+++ b/glib/gmain.c
@@ -3608,6 +3608,7 @@ g_main_context_release (GMainContext *context)
       g_critical ("g_main_context_release() called on a context (%p, owner %p, "
                   "owner count %u) which is not acquired by the current thread",
                   context, context_owner, context_owner_count);
+      return;
     }
 #endif  /* !G_DISABLE_CHECKS */
 
-- 
2.39.2

